gtest/src/system.rs

```diff
@@ -62,10 +66,53 @@ impl System {
         self.0.borrow_mut().block_info.timestamp += amount as u64;
     }
 
-    pub fn get_program<ID: Into<ProgramIdWrapper>>(&'_ self, id: ID) -> Program<'_> {
-        Program {
-            manager: &self.0,
-            id: id.into().0,
-        }
+    /// Gets a program by `id`
+    ///
+    /// Returns `None` if program with `id` doesn't exist or is a dormant actor.
+    pub fn get_program<ID: Into<ProgramIdWrapper>>(&'_ self, id: ID) -> Option<Program<'_>> {
+        let id = id.into().0;
+        self.0
+            .borrow()
+            .actors
+            .get(&id)
+            .and_then(|(actor, _)| match actor {
+                Actor::Dormant => None,
+                _ => Some(Program {
+                    id,
+                    manager: &self.0,
+                }),
+            })
+    }
```

Comment:
I would like to keep panic here for api-convenience

Reply:
discussed with @LouiseMedova and @breathx . Will remove option